components/opponent/commons/opponent.lua

```diff
@@ -262,6 +262,28 @@ function Opponent.resolve(opponent, date, options)
 	return opponent
 end
 
+--[[
+options.syncTeam: Whether to fetch team of the respectiveplayer information from variables or LPDB. Disabled by default.
+]]
+function Opponent.resolve(opponent, date, options)
+	options = options or {}
+	if opponent.type == Opponent.team then
+		opponent.template = TeamTemplate.resolve(opponent.template, date) or opponent.template or 'tbd'
+	elseif Opponent.typeIsParty(opponent.type) then
+		for _, player in ipairs(opponent.players) do
+			if options.syncTeam then
+				PlayerExt.syncTeam(player, {savePageVar = not Opponent.playerIsTbd(player)})
+			else
+				PlayerExt.populatePageName(player)
+			end
+			if player.team then
+				player.team = TeamTemplate.resolve(player.team, date)
+			end
+		end
+	end
+	return opponent
+end
```

Comment:
There's already a Opponent resolve function. This would override it. Add it to the existing

Reply:
> There's already a Opponent resolve function. This would override it. Add it to the existing

Got it, so I had the 1st revision up, I'm not sure where I'd missed if the syncteam needs to be elseif or not